src/core/eme/eme_manager.ts

```diff
@@ -148,23 +158,24 @@ export default function EMEManager(
                                        data: initData } });
       }
 
-      if (handledInitData.get(initDataType, initData) === true) {
+      if (!handledInitData.storeIfNone(initData, initDataType, true)) {
         log.debug("EME: Init data already received. Skipping it.");
         return observableOf({ type: "init-data-ignored" as const,
                               value: { type: initDataType, data: initData } });
       }
-      handledInitData.set(initDataType, initData, true);
 
       const session$ = getSession(encryptedEvent, mediaKeysInfos)
-        .pipe(map((evt) => ({
-          type: evt.type,
-          value: { initData: evt.value.initData,
-                   initDataType: evt.value.initDataType,
-                   mediaKeySession: evt.value.mediaKeySession,
-                   sessionType: evt.value.sessionType,
-                   keySystemOptions: mediaKeysInfos.keySystemOptions,
-                   sessionStorage: mediaKeysInfos.sessionStorage },
-        })));
+        .pipe(map((evt) => {
+          if (evt.type === "cleaning-old-session") {
```

Comment:
Is there a reason why you don't wait for `cleaned-old-session` ?

Reply:
Isn't there a chance that, if cleaning fails, the session may actually not be removed ?

Reply:
> Is there a reason why you don't wait for cleaned-old-session ?
I hesitated to do that, but what will happen if the initialization data is received while we are cleaning an old session?

I think it's better to remove the initData as soon as we decide to close the session. At best, the `LoadedSessionStore` could do any MediaKeySession creation only after the current one finished to be closed (not done now).

> Isn't there a chance that, if cleaning fails, the session may actually not be removed ?
Hm, that's a good question. Does it fail if two MediaKeySession are created for the same initialization data? What would be the resulting behavior?

For me, it's always difficult to handle errors when closing a MediaKeySession. You never know in which state it will be left in. Just creating another one might be the best solution, no?

Reply:
Ok i understand all of this.